deadlines/src/main/java/com/palantir/deadlines/Deadlines.java

```diff
@@ -246,4 +257,14 @@ Duration asDuration() {
             return valueNanos <= 0 ? Duration.ZERO : Duration.ofNanos(valueNanos);
         }
     }
+
+    interface DeadlineClock {
```

Comment:
Super pedantic naming nit: Since this is nested in `Deadlines`, I'd opt to name this `Clock` since it can be referenced as `Deadlines.Clock`. Doesn't really make much difference in practice though, especially since this is all package private api!